op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -0,0 +1,456 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-service/jsonutil"
+	"github.com/mattn/go-isatty"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+var (
+	deployConfigFlag = &cli.PathFlag{
+		Name:     "deploy-config",
+		Usage:    "Path to deploy config file",
+		Required: true,
+	}
+	l1DeploymentsFlag = &cli.PathFlag{
+		Name:  "l1-deployments",
+		Usage: "Path to L1 deployments JSON file. Cannot be used with --deployment-dir",
+	}
+	l1RPCFlag = &cli.StringFlag{
+		Name:  "l1-rpc",
+		Usage: "RPC URL for an Ethereum L1 node. Cannot be used with --l1-starting-block",
+	}
+	outfileL2Flag = &cli.PathFlag{
+		Name:  "outfile.l2",
+		Usage: "Path to L2 genesis output file",
+	}
+	outfileRollupFlag = &cli.PathFlag{
+		Name:  "outfile.rollup",
+		Usage: "Path to rollup output file",
+	}
+
+	dbPathFlag = &cli.StringFlag{
+		Name:     "db-path",
+		Usage:    "Path to database",
+		Required: true,
+	}
+	dbCacheFlag = &cli.IntFlag{
+		Name:  "db-cache",
+		Usage: "LevelDB cache size in mb",
+		Value: 1024,
+	}
+	dbHandlesFlag = &cli.IntFlag{
+		Name:  "db-handles",
+		Usage: "LevelDB number of handles",
+		Value: 60,
+	}
+	dryRunFlag = &cli.BoolFlag{
+		Name:  "dry-run",
+		Usage: "Dry run the upgrade by not committing the database",
+	}
+
+	flags = []cli.Flag{
+		deployConfigFlag,
+		l1DeploymentsFlag,
+		l1RPCFlag,
+		outfileL2Flag,
+		outfileRollupFlag,
+		dbPathFlag,
+		dbCacheFlag,
+		dbHandlesFlag,
+		dryRunFlag,
+	}
+
+	// from `packages/contracts-bedrock/deploy-config/internal-devnet.json`
+	EIP1559Denominator = uint64(50) // TODO(pl): select values
+	EIP1559Elasticity  = uint64(10)
```

Comment:
Are we selecting these now?

Reply:
I added this task as a follow-up in https://github.com/celo-org/optimism/issues/108

In general, it's one of the parameters we need to decide on. Is there a tracking issue for those already?